### DISCUSSION TITLE
How might I use my wasm code synchronously?

### DISCUSSION THREAD

**Discussion by akprasad** [wasm-bindgen/wasm-bindgen#3757] [Q&A] [answered]
### Summary

Thank you for this wonderful project.

I'm porting a JS library to Rust + Wasm. This library consists of simple synchronous functions, and I'm porting so that I have a single implementation that I can bind to multiple languages.

The stumbling block I'm hitting is that to use this port, I have to call `await init()`. This is unfortunate for a few reasons:

- Many of my users are not technically sophisticated, and they expect that adding the library with `<script src="library.js">` will be enough to start using it.

- Since my original JS library is simple synchronous code, I'd like to make this port as close to a drop-in as I can.

- I'm still debugging this (and I'm not a sophisticated JS user, so maybe there's a simple fix), but using `await init()` is causing my library to load after some other setup code in my project, which means my app isn't being initialized properly. My code is simple JS, no Node / Webpack / etc.

I'm curious how I might work around this, with an ideal outcome being some form of simple synchronous behavior.

**Comment by daxpedda** [comment, marked as answer]
We do export a function called `initSync` which you could use instead of the default.
But you have to somehow fetch the Wasm file which is usually asynchronous by nature.

See the [example](https://github.com/rustwasm/wasm-bindgen/tree/c0ee195d7ebbb18b8517d115c4f3bfee4b797a3d/examples/synchronous-instantiation).

**Reply by akprasad** [reply]
Thank you for your reply! I had seen `initSync` in the docs but didn't know you had a worked-out example as well. Thanks, I'll take a look and follow up here if I run into any trouble.
